### PR TITLE
[DDPMScheduler] Load `alpha_cumprod` to device to avoid redundant data movement.

### PR DESCRIPTION
**Pull request by woshiyyya**
# What does this PR do?

In my stable diffusion training workload, I am adding noise to the input image latents at each training step. From some analysis on the flamegraph, it seems that the `self.alpha_cumprod.to` operation in `DDPMScheduler.add_noise` takes a lot of time and becomes a bottleneck. 

This PR moves the tensor to the sample's device at the first time, then the `.to` operations in the following `add_noise` calls will be noop. The flamegraph after this change indicates that `add_noise` calls take much less time.

This might not be the most elegant solution, but it did reduce a huge overhead.

Before:
Screenshot 2024-01-23 at 4 30 05 PM

After:
Screenshot 2024-01-24 at 4 05 00 PM

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by woshiyyya** [comment]
cc @dhruvrnaik @yiyixuxu @sayakpaul

File: src/diffusers/schedulers/scheduling_ddpm.py [hunk 1]
```diff
@@ -503,7 +503,10 @@ def add_noise(
         timesteps: torch.IntTensor,
     ) -> torch.FloatTensor:
         # Make sure alphas_cumprod and timestep have same device and dtype as original_samples
-        alphas_cumprod = self.alphas_cumprod.to(device=original_samples.device, dtype=original_samples.dtype)
+        # Move the self.alphas_cumprod to device to avoid redundant CPU to GPU data movement 
+        # for the subsequent add_noise calls
+        self.alphas_cumprod = self.alphas_cumprod.to(device=original_samples.device)
+        alphas_cumprod = self.alphas_cumprod.to(dtype=original_samples.dtype)
```
Comment (sayakpaul) [inline src/diffusers/schedulers/scheduling_ddpm.py:509]: I am personally okay with this.

**Comment by sayakpaul** [comment]
Be a little patient as @yiyixuxu gets to this. But we will, for sure.

**Review by yiyixuxu — COMMENTED** [review]
great, thanks! 
can we fix the test? will merge once tests pass :)

**Comment by woshiyyya** [comment]
Ok let me try to fix them:)

**Comment by yiyixuxu** [comment]
@woshiyyya need to run `make fix-copies`

**Pull request merged**